{"cvrfdoc":{"document_title":"Red Hat Security Advisory: rh-python38 security, bug fix, and enhancement update","document_distribution":"Copyright Â© 2021 Red Hat, Inc. All rights reserved.","document_references":{"reference":[{"description":"https://access.redhat.com/errata/RHSA-2021:3254","type":"Self","url":"https://access.redhat.com/errata/RHSA-2021:3254"},{"description":"https://access.redhat.com/security/updates/classification/#moderate","type":"External","url":"https://access.redhat.com/security/updates/classification/#moderate"},{"description":"https://access.redhat.com/articles/5860431","type":"External","url":"https://access.redhat.com/articles/5860431"},{"description":"https://access.redhat.com/documentation/en-us/red_hat_software_collections/3/html-single/3.7_release_notes/index#sect-RHSCL-Other-notes","type":"External","url":"https://access.redhat.com/documentation/en-us/red_hat_software_collections/3/html-single/3.7_release_notes/index#sect-RHSCL-Other-notes"}]},"aggregate_severity":"Moderate","document_tracking":{"initial_release_date":"2021-08-24T12:50:00Z","identification":{"id":"RHSA-2021:3254"},"revision_history":{"revision":{"date":"2021-11-10T21:21:00Z","number":"1","description":"Current version"}},"generator":{"date":"2021-11-13T21:15:00Z","engine":"Red Hat SDEngine 3.3.0"},"current_release_date":"2021-11-10T21:21:00Z","version":"1","status":"Final"},"product_tree":{"relationship":[{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-babel-0:2.7.0-12.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-babel-0:2.7.0-12.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-0:3.8.11-2.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-0:3.8.11-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-cryptography-0:2.8-5.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-cryptography-0:2.8-5.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-jinja2-0:2.10.3-6.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-jinja2-0:2.10.3-6.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-lxml-0:4.4.1-7.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-lxml-0:4.4.1-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-pip-0:19.3.1-2.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-pip-0:19.3.1-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7","product_reference":"rh-python38-python-urllib3-0:1.25.7-7.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-urllib3-0:1.25.7-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-babel-0:2.7.0-12.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-babel-0:2.7.0-12.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-0:3.8.11-2.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-0:3.8.11-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-cryptography-0:2.8-5.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-cryptography-0:2.8-5.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-jinja2-0:2.10.3-6.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-jinja2-0:2.10.3-6.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-lxml-0:4.4.1-7.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-lxml-0:4.4.1-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-pip-0:19.3.1-2.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-pip-0:19.3.1-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Server-RHSCL-3.7-7.7.Z","product_reference":"rh-python38-python-urllib3-0:1.25.7-7.el7","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-urllib3-0:1.25.7-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-babel-0:2.7.0-12.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-babel-0:2.7.0-12.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-0:3.8.11-2.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-0:3.8.11-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-cryptography-0:2.8-5.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-cryptography-0:2.8-5.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-jinja2-0:2.10.3-6.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-jinja2-0:2.10.3-6.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-lxml-0:4.4.1-7.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-lxml-0:4.4.1-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-pip-0:19.3.1-2.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-pip-0:19.3.1-2.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"},{"relates_to_product_reference":"7Workstation-RHSCL-3.7","product_reference":"rh-python38-python-urllib3-0:1.25.7-7.el7","full_product_name":{"product_id":"7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-urllib3-0:1.25.7-7.el7 as a component of Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"relation_type":"Default Component Of"}],"branch":[{"name":"Red Hat Software Collections","type":"Product Family","branch":[{"name":"Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)","full_product_name":{"product_id":"7Server-RHSCL-3.7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"Red Hat Software Collections for Red Hat Enterprise Linux Server (v. 7)"},"type":"Product Name"},{"name":"Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)","full_product_name":{"product_id":"7Server-RHSCL-3.7-7.7.Z","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"Red Hat Software Collections for Red Hat Enterprise Linux Server EUS (v. 7.7)"},"type":"Product Name"},{"name":"Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)","full_product_name":{"product_id":"7Workstation-RHSCL-3.7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"Red Hat Software Collections for Red Hat Enterprise Linux Workstation (v. 7)"},"type":"Product Name"}]},{"name":"rh-python38-babel-0:2.7.0-12.el7","full_product_name":{"product_id":"rh-python38-babel-0:2.7.0-12.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-babel-2.7.0-12.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-0:3.8.11-2.el7","full_product_name":{"product_id":"rh-python38-python-0:3.8.11-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-3.8.11-2.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-cryptography-0:2.8-5.el7","full_product_name":{"product_id":"rh-python38-python-cryptography-0:2.8-5.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-cryptography-2.8-5.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-jinja2-0:2.10.3-6.el7","full_product_name":{"product_id":"rh-python38-python-jinja2-0:2.10.3-6.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-jinja2-2.10.3-6.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-lxml-0:4.4.1-7.el7","full_product_name":{"product_id":"rh-python38-python-lxml-0:4.4.1-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-lxml-4.4.1-7.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-pip-0:19.3.1-2.el7","full_product_name":{"product_id":"rh-python38-python-pip-0:19.3.1-2.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-pip-19.3.1-2.el7.src.rpm"},"type":"Product Version"},{"name":"rh-python38-python-urllib3-0:1.25.7-7.el7","full_product_name":{"product_id":"rh-python38-python-urllib3-0:1.25.7-7.el7","cpe":"cpe:/a:redhat:rhel_software_collections:3::el7","product_name":"rh-python38-python-urllib3-1.25.7-7.el7.src.rpm"},"type":"Product Version"}]},"document_publisher":{"issuing_authority":"Red Hat Product Security","contact_details":"secalert@redhat.com","type":"Vendor"},"vulnerability":[{"notes":{"note":"python-cryptography: Bleichenbacher timing oracle attack against RSA decryption"},"cve":"CVE-2020-25659","references":{"reference":[{"description":"https://cryptography.io/en/latest/changelog.html#v3-2","url":"https://cryptography.io/en/latest/changelog.html#v3-2"},{"description":"CVE-2020-25659","url":"https://access.redhat.com/security/cve/CVE-2020-25659"},{"description":"bz#1889988: CVE-2020-25659 python-cryptography: Bleichenbacher timing oracle attack against RSA decryption","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1889988"}]},"release_date":"2020-10-25T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"acknowledgments":{"acknowledgment":{"description":"This issue was discovered by Hubert Kario (Red Hat)."}},"discovery_date":"2020-10-21T00:00:00Z","ordinal":"1"},{"notes":{"note":"python: Unsafe use of eval() on data retrieved via HTTP in the test suite"},"cve":"CVE-2020-27619","references":{"reference":[{"description":"CVE-2020-27619","url":"https://access.redhat.com/security/cve/CVE-2020-27619"},{"description":"bz#1889886: CVE-2020-27619 python: Unsafe use of eval() on data retrieved via HTTP in the test suite","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1889886"}]},"release_date":"2020-10-05T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-10-08T00:00:00Z","ordinal":"2"},{"notes":{"note":"python-lxml: mXSS due to the use of improper parser"},"cve":"CVE-2020-27783","references":{"reference":[{"description":"CVE-2020-27783","url":"https://access.redhat.com/security/cve/CVE-2020-27783"},{"description":"bz#1901633: CVE-2020-27783 python-lxml: mXSS due to the use of improper parser","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1901633"}]},"release_date":"2020-10-18T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2020-11-25T00:00:00Z","ordinal":"3"},{"notes":{"note":"python-jinja2: ReDoS vulnerability in the urlize filter"},"cve":"CVE-2020-28493","references":{"reference":[{"description":"CVE-2020-28493","url":"https://access.redhat.com/security/cve/CVE-2020-28493"},{"description":"bz#1928707: CVE-2020-28493 python-jinja2: ReDoS vulnerability in the urlize filter","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1928707"}]},"release_date":"2021-02-01T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-02-15T00:00:00Z","ordinal":"4"},{"notes":{"note":"python-cryptography: Large inputs for symmetric encryption can trigger integer overflow leading to buffer overflow"},"cve":"CVE-2020-36242","references":{"reference":[{"description":"https://cryptography.io/en/latest/changelog.html#v3-3-2","url":"https://cryptography.io/en/latest/changelog.html#v3-3-2"},{"description":"CVE-2020-36242","url":"https://access.redhat.com/security/cve/CVE-2020-36242"},{"description":"bz#1926226: CVE-2020-36242 python-cryptography: Large inputs for symmetric encryption can trigger integer overflow leading to buffer overflow","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1926226"}]},"release_date":"2020-12-09T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-02-07T00:00:00Z","ordinal":"5"},{"notes":{"note":"python: Stack-based buffer overflow in PyCArg_repr in _ctypes/callproc.c"},"cve":"CVE-2021-3177","references":{"reference":[{"description":"CVE-2021-3177","url":"https://access.redhat.com/security/cve/CVE-2021-3177"},{"description":"bz#1918168: CVE-2021-3177 python: Stack-based buffer overflow in PyCArg_repr in _ctypes/callproc.c","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1918168"}]},"release_date":"2021-01-19T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-01-20T00:00:00Z","ordinal":"6"},{"notes":{"note":"python: Information disclosure via pydoc"},"cve":"CVE-2021-3426","references":{"reference":[{"description":"CVE-2021-3426","url":"https://access.redhat.com/security/cve/CVE-2021-3426"},{"description":"bz#1935913: CVE-2021-3426 python: Information disclosure via pydoc","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1935913"}]},"release_date":"2021-03-10T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"acknowledgments":{"acknowledgment":{"description":"Red Hat would like to thank David SchwÃ¶rer (Fedora) for reporting this issue."}},"discovery_date":"2021-01-19T00:00:00Z","ordinal":"7"},{"notes":{"note":"python-pip: Incorrect handling of unicode separators in git references"},"cve":"CVE-2021-3572","references":{"reference":[{"description":"CVE-2021-3572","url":"https://access.redhat.com/security/cve/CVE-2021-3572"},{"description":"bz#1962856: CVE-2021-3572 python-pip: Incorrect handling of unicode separators in git references","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1962856"}]},"release_date":"2021-04-24T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Low","type":"Impact"}},"discovery_date":"2021-05-19T00:00:00Z","ordinal":"8"},{"notes":{"note":"python: urllib: Regular expression DoS in AbstractBasicAuthHandler"},"cve":"CVE-2021-3733","references":{"reference":[{"description":"https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-14-final","url":"https://docs.python.org/3.6/whatsnew/changelog.html#python-3-6-14-final"},{"description":"https://docs.python.org/3.7/whatsnew/changelog.html#python-3-7-11-final","url":"https://docs.python.org/3.7/whatsnew/changelog.html#python-3-7-11-final"},{"description":"https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-10-final","url":"https://docs.python.org/3.8/whatsnew/changelog.html#python-3-8-10-final"},{"description":"https://docs.python.org/3.9/whatsnew/changelog.html#python-3-9-5-final","url":"https://docs.python.org/3.9/whatsnew/changelog.html#python-3-9-5-final"},{"description":"CVE-2021-3733","url":"https://access.redhat.com/security/cve/CVE-2021-3733"},{"description":"bz#1995234: CVE-2021-3733 python: urllib: Regular expression DoS in AbstractBasicAuthHandler","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1995234"}]},"release_date":"2021-08-10T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-08-10T00:00:00Z","ordinal":"9"},{"notes":{"note":"python-babel: Relative path traversal allows attacker to load arbitrary locale files and execute arbitrary code"},"cve":"CVE-2021-20095","references":{"reference":[{"description":"https://www.tenable.com/security/research/tra-2021-14","url":"https://www.tenable.com/security/research/tra-2021-14"},{"description":"CVE-2021-20095","url":"https://access.redhat.com/security/cve/CVE-2021-20095"},{"description":"bz#1955615: CVE-2021-20095 CVE-2021-42771 python-babel: Relative path traversal allows attacker to load arbitrary locale files and execute arbitrary code","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1955615"}]},"release_date":"2021-04-28T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-04-29T00:00:00Z","ordinal":"10"},{"notes":{"note":"python: Web cache poisoning via urllib.parse.parse_qsl and urllib.parse.parse_qs by using a semicolon in query parameters"},"cve":"CVE-2021-23336","references":{"reference":[{"description":"https://snyk.io/vuln/SNYK-UPSTREAM-PYTHONCPYTHON-1074933","url":"https://snyk.io/vuln/SNYK-UPSTREAM-PYTHONCPYTHON-1074933"},{"description":"CVE-2021-23336","url":"https://access.redhat.com/security/cve/CVE-2021-23336"},{"description":"bz#1928904: CVE-2021-23336 python: Web cache poisoning via urllib.parse.parse_qsl and urllib.parse.parse_qs by using a semicolon in query parameters","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1928904"}]},"release_date":"2021-02-13T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-02-15T00:00:00Z","ordinal":"11"},{"notes":{"note":"python-lxml: Missing input sanitization for formaction HTML5 attributes may lead to XSS"},"cve":"CVE-2021-28957","references":{"reference":[{"description":"CVE-2021-28957","url":"https://access.redhat.com/security/cve/CVE-2021-28957"},{"description":"bz#1941534: CVE-2021-28957 python-lxml: Missing input sanitization for formaction HTML5 attributes may lead to XSS","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1941534"}]},"release_date":"2021-03-21T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-03-21T00:00:00Z","ordinal":"12"},{"notes":{"note":"python-ipaddress: Improper input validation of octal strings"},"cve":"CVE-2021-29921","references":{"reference":[{"description":"https://python-security.readthedocs.io/vuln/ipaddress-ipv4-leading-zeros.html","url":"https://python-security.readthedocs.io/vuln/ipaddress-ipv4-leading-zeros.html"},{"description":"CVE-2021-29921","url":"https://access.redhat.com/security/cve/CVE-2021-29921"},{"description":"bz#1957458: CVE-2021-29921 python-ipaddress: Improper input validation of octal strings","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1957458"}]},"release_date":"2021-04-30T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-04-30T00:00:00Z","ordinal":"13"},{"notes":{"note":"python-urllib3: ReDoS in the parsing of authority part of URL"},"cve":"CVE-2021-33503","references":{"reference":[{"description":"https://github.com/advisories/GHSA-q2q7-5pp4-w6pg","url":"https://github.com/advisories/GHSA-q2q7-5pp4-w6pg"},{"description":"CVE-2021-33503","url":"https://access.redhat.com/security/cve/CVE-2021-33503"},{"description":"bz#1968074: CVE-2021-33503 python-urllib3: ReDoS in the parsing of authority part of URL","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1968074"}]},"release_date":"2021-06-01T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-06-01T00:00:00Z","ordinal":"14"},{"notes":{"note":"python-babel: Relative path traversal allows attacker to load arbitrary locale files and execute arbitrary code"},"cve":"CVE-2021-42771","references":{"reference":[{"description":"https://www.tenable.com/security/research/tra-2021-14","url":"https://www.tenable.com/security/research/tra-2021-14"},{"description":"CVE-2021-42771","url":"https://access.redhat.com/security/cve/CVE-2021-42771"},{"description":"bz#1955615: CVE-2021-20095 CVE-2021-42771 python-babel: Relative path traversal allows attacker to load arbitrary locale files and execute arbitrary code","url":"https://bugzilla.redhat.com/show_bug.cgi?id=1955615"}]},"release_date":"2021-04-28T00:00:00Z","involvements":{"involvement":{"party":"Vendor","status":"Completed"}},"product_statuses":{"status":{"product_id":["7Server-RHSCL-3.7-7.7.Z:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7-7.7.Z:rh-python38-python-urllib3-0:1.25.7-7.el7","7Server-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Server-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Server-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Server-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Server-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Server-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Server-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7","7Workstation-RHSCL-3.7:rh-python38-babel-0:2.7.0-12.el7","7Workstation-RHSCL-3.7:rh-python38-python-0:3.8.11-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-cryptography-0:2.8-5.el7","7Workstation-RHSCL-3.7:rh-python38-python-jinja2-0:2.10.3-6.el7","7Workstation-RHSCL-3.7:rh-python38-python-lxml-0:4.4.1-7.el7","7Workstation-RHSCL-3.7:rh-python38-python-pip-0:19.3.1-2.el7","7Workstation-RHSCL-3.7:rh-python38-python-urllib3-0:1.25.7-7.el7"],"type":"Fixed"}},"remediations":{"remediation":{"description":"For details on how to apply this update, which includes the changes described in this advisory, refer to:\nhttps://access.redhat.com/articles/11258","type":"Vendor Fix","url":"https://access.redhat.com/errata/RHSA-2021:3254"}},"threats":{"threat":{"description":"Moderate","type":"Impact"}},"discovery_date":"2021-04-29T00:00:00Z","ordinal":"15"}],"document_notes":{"note":["An update for rh-python38-babel, rh-python38-python, rh-python38-python-cryptography, rh-python38-python-jinja2, rh-python38-python-lxml, rh-python38-python-pip, and rh-python38-python-urllib3 is now available for Red Hat Software Collections.\nRed Hat Product Security has rated this update as having a security impact of Moderate. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE link(s) in the References section.","Python is an interpreted, interactive, object-oriented programming language, which includes modules, classes, exceptions, very high level dynamic data types and dynamic typing. Python supports interfaces to many system calls and libraries, as well as to various windowing systems.\nThe following packages have been upgraded to a later upstream version: rh-python38-python (3.8.11). (BZ#1969523)\nSecurity Fix(es):\n* python-cryptography: Bleichenbacher timing oracle attack against RSA decryption (CVE-2020-25659)\n* python: Unsafe use of eval() on data retrieved via HTTP in the test suite (CVE-2020-27619)\n* python-lxml: mXSS due to the use of improper parser (CVE-2020-27783)\n* python-jinja2: ReDoS vulnerability due to the sub-pattern (CVE-2020-28493)\n* python-cryptography: Large inputs for symmetric encryption can trigger integer overflow leading to buffer overflow (CVE-2020-36242)\n* python: Stack-based buffer overflow in PyCArg_repr in _ctypes/callproc.c (CVE-2021-3177)\n* python: Information disclosure via pydoc (CVE-2021-3426)\n* python-babel: Relative path traversal allows attacker to load arbitrary locale files and execute arbitrary code (CVE-2021-20095)\n* python: Web cache poisoning via urllib.parse.parse_qsl and urllib.parse.parse_qs by using a semicolon in query parameters (CVE-2021-23336)\n* python-lxml: Missing input sanitization for formaction HTML5 attributes may lead to XSS (CVE-2021-28957)\n* python-ipaddress: Improper input validation of octal strings (CVE-2021-29921)\n* python-urllib3: ReDoS in the parsing of authority part of URL (CVE-2021-33503)\n* python-pip: Incorrect handling of unicode separators in git references (CVE-2021-3572)\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\nAdditional information\n* Changes in the default separator for the Python urllib parsing functions\nTo mitigate the Web Cache Poisoning CVE-2021-23336 in the Python urllib library, the default separator for the urllib.parse.parse_qsl and urllib.parse.parse_qs functions is being changed from both ampersand (&) and semicolon (;) to only an ampersand.\nThe change of the default separator is potentially backwards incompatible, therefore Red Hat provides a way to configure the behavior in Python packages where the default separator has been changed. In addition, the affected urllib parsing functions issue a warning if they detect that a customerâs application has been affected by the change.\nFor more information, see the Knowledgebase article \"Mitigation of Web Cache Poisoning in the Python urllib library (CVE-2021-23336)\" linked from the References section.\n* The Python \"ipaddress\" module no longer allows leading zeros in IPv4 addresses\nTo mitigate CVE-2021-29921, the Python \"ipaddress\" module now rejects IPv4 addresses with leading zeros with an \"AddressValueError: Leading zeros are not permitted\" error.\nCustomers who rely on the previous behavior can pre-process their IPv4 address inputs to strip the leading zeros off. For details, see the Red Hat Software Collections 3.7 Release Notes.","This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original."]},"document_type":"Security Advisory"}}